components/eam/src/chemistry/utils/modal_aero_deposition.F90

```diff
@@ -332,26 +371,51 @@ subroutine set_srf_wetdep(aerdepwetis, aerdepwetcw, cam_out)
    !        dry deposition fluxes are positive into surface.
    !        srf models want positive definite fluxes.
    do i = 1, ncol
-
       ! black carbon fluxes
-      if (idx_bc1>0) &
-         cam_out%bcphiwet(i) = cam_out%bcphiwet(i) -(aerdepwetis(i,idx_bc1)+aerdepwetcw(i,idx_bc1))
-      if (idx_bc4>0) &
-         cam_out%bcphiwet(i) = cam_out%bcphiwet(i) -(aerdepwetis(i,idx_bc4)+aerdepwetcw(i,idx_bc4))
+      bcphiwet_sum = 0.0_r8
+      do ispec = 1, nbc
+         if (idx_bc1(1)>0) then
+            do ispec = 1, nbc
```

Comment:
Something weird is going on here, ispec should not be redefined inside the ispec loop.

Reply:
Agreed. That looks like a typo when porting over the changes. I recall this module having a different format that E3SMv1 which made this adaption pretty finnicky. The inner ispec loop shouldn't be there as per the organic example below (lines 391-409).